Recruit-Training/Embedded-Recruit-Training/Core/Src/BlueLedToggleTask.c

```diff
@@ -9,8 +12,39 @@ void blueLedToggleTask(void const* arg)
 
     for (;;)
     {
-        osDelayUntil(prevWakeTime + 1);  //TODO: Replace 1 with time period for delay
+        osDelayUntil(prevWakeTime + BLUE_LED_TOGGLE_FREQ);  //TODO: Replace 1 with time period for delay
+        
         //TODO: Check blue toggle flag and toggle blue LED
+        if (blueToggleFlag)
+        {
+            HAL_GPIO_TogglePin(GPIOA, GPIO_PIN_5);
+        }
+        
+        GPIO_PinState blueStatus = HAL_GPIO_ReadPin(GPIOA, GPIO_PIN_5);
+
         //TODO: Send CAN message indicating current state of LED
+        while (osMutexAcquire(canMutex, 0) != osOK) {}
```

Comment:
We usually don't loop to wait for the mutex to be acquired. If the status `!= osOK` we would just `continue`.
```suggestion
        if (osMutexAcquire(canMutex, 0) != osOK) continue;
```